### PR TITLE
feat: added download feature along with other improvements.

### PR DESCRIPTION
**Pull request by shub78910**

### CODE REVIEW

**Comment by itsmais** [comment]
Hello @shub78910 ! 

Thank you so much for adding the download option. I tried it, and it works perfectly!

I have a couple of comments for you:

**1-** I see that you changed the quotes from single to double. I'm not sure that's quite necessary since it's an editor/formatter choice. It would be nice if we could revert those changes.

**2-** I noticed that on the settings page, where the download button is, the next is still `"Copy tasks data to clipboard"`. Maybe we should change it to something like `"Download data as .txt"`.

**3-** There's a merge conflict that prevents me from merging your work. Could you please resolve it?

**To summarize, if you think double quotes are the way to go, I don't really mind. Regarding the second point, I can take care of it later if you want as well (I noticed that I also need to set a font size for the button class). 
What is most important now is resolving the merge conflict in this PR.**

Thank you! Please let me know if something is unclear.

**Comment by shub78910** [comment]
Hey, the suggestions were great.
I have tried to resolve all the conflicts, and most of them were because of the loader addition. 
I have included the loader code and have tried to solve the conflicts.
Coming to the quotes part, my editor has a default of double quotes, and I believe we can let that slide.
Also, I have changed the button to "download as text" instead of copy.
Hope these are good enough. Let me know

**Comment by itsmais** [comment]
Hello @shub78910 !
I noticed that the content of the downloaded file (`tasks.txt`) are not what we want the first time you try to download after you add a new task. 
Here's how the file is looking like:
image

Any idea why this might be happening? Let me know if you need help reproducing the error.

**Comment by itsmais** [comment]
I see why you commented it. I'll see how we can re-add it later without issues. 

I tested the download functionality and it's working fine now 🚀 I'm merging this and closing the issue. 

Thank you so much for contributing to Time It! Hope to see you soon 👋
